src/MatrixManifold.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    MatrixManifold{M <: Manifold} <: Manifold
+
+A manifold to encapsulate manifolds working on array representations of
+`MPoints` and `TVectors` in a transparent way, such that for these manifolds its
+not necessary to introduce explicit types for the points and tangent vectors,
+but they are encapusalted/stripped automatically when needed.
+"""
+struct MatrixManifold{M <: Manifold}
+    manifold::M
+end
+convert(::Type{M},m::MatrixManifold{M}) where M <: Manifold = m.manifold
+convert(::Type{MatrixManifold{M}},m::M) where M <: Manifold = MatrixManifold(M)
+
+struct MatrixMPoint{V <: AbstractVector{<:Real}} <: MPoint
```

Comment:
Perhaps I'm misunderstanding what these are for, but why is the wrapped array of a matrix point forced to be a vector?

Also, do we want to only support real matrix points? What about complex matrix groups like SU(2)?

Reply:
Well it's start to think about. Yes Complex would be a good idea – Oh and I meant to always take AbstractArrays – I am sorry, I was maybe not careful enough in writing.